### ISSUE TITLE
508-defect-2 Mapbox in keyboard user flow

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-team#16971]
## Issue Description
Zoom in/out to Search this area button focus management is inconsistent

On Chrome, if a user fires a zoom in/out event for the first time using space or enter, the Search this area button will appear and receive focus. This behavior is acceptable from an accessibility standpoint, but from a UX perspective it can be argued that this is taking control away from the user. e.g. What if a user wants to zoom out several times before conducting a search? A polite notification that the option to search this area has appeared OR making the search this area button always persistent may address this.
On Safari/VoiceOver, the space or enter keys were inconsistent in moving focus to the Search this area button. Although focus sometimes would not be moved, the screen reader still announced the Search this area button as though focus had been moved. When compounded with the fact that changes of context when zooming are not announced, a user may continue hitting the zoom in/out buttons thinking they are searching the area when in reality they're still on the zoom in/out buttons.
On all browsers, the search this area button is not reachable in the tab order. It is only accessible when focus is moved to it on the first time the user fires a zoom in or out.

---
## Tasks
- [ ] Work with accessibility team members to determine ideal flow for focus. 
- [ ] Add appropriate role for interactive content for Mapbox icon  
- [ ] Screen reader users can hear the attribution content separately from the map interactive controls. 
- [ ] Identify any legal policies related to Mapbox attribution
- [ ] Review 508-defect-2 [SCREENREADER]: The map is announcing a lot of extra data on focus and each droplet needs a usable name #515 to inform handling of pins

## Acceptance Criteria (TBD with input from Accessibility team)
**After the user interacts with the map ->**
- [ ] All map elements receive focus
      - "search this area" button
      - location pins
      - zoom controls
      - Mapbox attribution
      - Mapbox info button
- [ ] Screenreaders announce all elements
- [ ] Focus has logical flow
- [ ] "Current location" marker does not need to be included in the focus flow (will confirm with Josh/accessibility team)

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `tools-be`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by mmiddaugh** [comment]
Per Josh, we should skip the Mapbox icon in the keyboard flow
Additional accessibility changes will likely be possible once we are able to evaluate/leverage the plug-in.

**Comment by mmiddaugh** [comment]
We must leave Mapbox icon in logical flow due to legal obligations

**Comment by mmiddaugh** [comment]
Keyboard user flow has been addressed
